fs/move.py

```diff
@@ -53,26 +58,54 @@ def move_file(
         dst_path (str): Path to a file on ``dst_fs``.
         preserve_time (bool): If `True`, try to preserve mtime of the
             resources (defaults to `False`).
+        cleanup_dest_on_error (bool): If `True`, tries to delete the file copied to
+            dst_fs if deleting the file from src_fs fails.
 
     """
-    with manage_fs(src_fs) as _src_fs:
-        with manage_fs(dst_fs, create=True) as _dst_fs:
+    with manage_fs(src_fs, writeable=True) as _src_fs:
+        with manage_fs(dst_fs, writeable=True, create=True) as _dst_fs:
             if _src_fs is _dst_fs:
                 # Same filesystem, may be optimized
                 _src_fs.move(
                     src_path, dst_path, overwrite=True, preserve_time=preserve_time
                 )
-            else:
-                # Standard copy and delete
-                with _src_fs.lock(), _dst_fs.lock():
-                    copy_file(
-                        _src_fs,
-                        src_path,
-                        _dst_fs,
-                        dst_path,
-                        preserve_time=preserve_time,
-                    )
+                return
+
+            if _src_fs.hassyspath(src_path) and _dst_fs.hassyspath(dst_path):
+                # if both filesystems have a syspath we create a new OSFS from a
+                # common parent folder and use it to move the file.
+                try:
+                    src_syspath = _src_fs.getsyspath(src_path)
+                    dst_syspath = _dst_fs.getsyspath(dst_path)
+                    common = commonpath([src_syspath, dst_syspath])
+                    rel_src = frombase(common, src_syspath)
+                    rel_dst = frombase(common, dst_syspath)
+                    with _src_fs.lock(), _dst_fs.lock():
+                        with open_fs(common, writeable=True) as base:
```

Comment:
Given that we know that `common` is a syspath, I guess we could just use the `OSFS` constructor directly here, and "bypass" the `open_fs` machinery?

Reply:
Absolutely, in particular because it's possible (really discouraged, but possible) to edit the default opener to something different by setting `fs.opener.registry.default_opener` to something else.